### ISSUE TITLE
sorting an array and getting the first value in it 

### ISSUE THREAD

**Issue by anandrajakumaran** [bazaarvoice/jolt#696]
Input JSON:
{
  "scores": [
    "P2P",
    "P2B",
    "P2C"
  ]
}

Output JSON:

I need to sort this array and get the first element in the sorted array.For the above input Json the sorted array would be {
  "scores": [
    "P2B",
    "P2C",
    "P2P"
  ]
}

and i would like to get the first value from it which is "P2B"

Tried with this spec but does not seem to work:
[
  {
    "operation": "modify-overwrite-beta",
    "spec": {
      "sortedScores": "=sort(@(1,scores))",
      // what is the length of the scores array
      "numScores": "=size(@(1,scores))",
      //
      // Pull individual data out of the scores array
      "firstScore": "=firstElement(@(1,scores))",
      "lastScore": "=lastElement(@(1,scores))",
      //
      // Assuming that the scores array is always size of 5
      "scoreAtMidPoint": "=elementAt(@(1,scores),2)"
        // Sort the scores ascending
    }
  }, {
    "operation": "modify-overwrite-beta",
    "spec": {
      // Pull individual data out of the scores array
      "sortedScores": "=firstElement(@(1,sortedScores))",
      "sortedScores1": "=lastElement(@(1,sortedScores))"
    }
  }
]

**Comment by milosimpson** [comment]
Yeah, so there seems to be a problem with the "sort", it doesn't make a copy of the incoming list, and it seems to mess it up so that it isn't usable after being sorted.

**Comment by milosimpson** [comment]
Yeah, I think I see the problem.   It takes in a Java List, but it returns a sorted primitive array.   Which the other functions "firstElement" etc don't understand.

**Comment by ianupmondal** [comment]
any solution for this issue?
